src/models/predicate/link_v02.rs

```diff
@@ -0,0 +1,155 @@
+use std::collections::BTreeMap;
+
+use serde_derive::{Deserialize, Serialize};
+
+use super::{PredicateLayout, PredicateVersion, PredicateWrapper};
+use crate::interchange::{DataInterchange, Json};
+use crate::models::byproducts::ByProducts;
+use crate::models::step::Command;
+use crate::models::{LinkMetadata, TargetDescription, VirtualTargetPath};
+use crate::Result;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(deny_unknown_fields)]
+/// Predicate `LinkV02` means the predicate of original compatible format.
+///
+/// [LinkV02](https://in-toto.io/Link/v0.2)
+/// can be used together with most states.
+pub struct LinkV02 {
+    name: String,
+    materials: BTreeMap<VirtualTargetPath, TargetDescription>,
+    env: Option<BTreeMap<String, String>>,
+    command: Command,
+    byproducts: ByProducts,
+}
+
+impl From<LinkMetadata> for LinkV02 {
```

Comment:
Is the V02 from the SLSA Provenance's version? Because that applies exclusively to provenance, not link metadata. We do need to version the Link spec but we can match the in-toto spec's versioning probably.

Reply:
The right conversion here should be `From<MetaData>`, but the MetaData cannot be used to fetch any status for its design, so I use LinkMetaData for temporary resource, thus the test can be engaged.

Will be rewrite to `MetaData` after the refactor of MetaData, relying https://github.com/in-toto/in-toto-rs/pull/34

Or maybe I comment out the code and corresponding test?